testing/base/source_gen.cpp

```diff
@@ -669,24 +893,28 @@ auto SourceGen::GenerateClassDef(const ClassParams& params,
 
   for ([[maybe_unused]] int _ : llvm::seq(params.private_function_decls)) {
     os << line_sep;
-    GenerateFunctionDecl(unique_member_names.Pop(), /*is_private=*/true,
-                         /*is_method=*/false,
-                         state.private_function_param_counts.pop_back_val(),
-                         /*indent=*/"  ", state.param_names, os);
+    GenerateFunctionDecl(
+        unique_member_names.Pop(), /*is_private=*/true,
+        /*is_method=*/false,
+        state.private_function_param_counts().pop_back_val(),
+        /*indent=*/"  ", state.param_names(),
+        [&] { return state.GetValidTypeName(); }, os);
   }
   for ([[maybe_unused]] int _ : llvm::seq(params.private_method_decls)) {
     os << line_sep;
-    GenerateFunctionDecl(unique_member_names.Pop(), /*is_private=*/true,
-                         /*is_method=*/true,
-                         state.private_method_param_counts.pop_back_val(),
-                         /*indent=*/"  ", state.param_names, os);
+    GenerateFunctionDecl(
+        unique_member_names.Pop(), /*is_private=*/true,
+        /*is_method=*/true, state.private_method_param_counts().pop_back_val(),
+        /*indent=*/"  ", state.param_names(),
+        [&] { return state.GetValidTypeName(); }, os);
   }
   os << line_sep;
-  for ([[maybe_unused]] int _ : llvm::seq(params.private_field_decls)) {
+  for (llvm::StringRef type_name : field_type_names) {
     if (!IsCpp()) {
```

Comment:
Perhaps swap this around so you don't have a negated condition?

Reply:
There are a bunch of negative conditions and positive ones mixed....

Can I address this one way or the other in a follow-up PR? Can chat to see which direction works best.